model/pdata/generated_metrics.go

```diff
@@ -1106,6 +1124,18 @@ func (ms NumberDataPoint) SetTimestamp(v Timestamp) {
 	(*ms.orig).TimeUnixNano = uint64(v)
 }
 
+// Type returns the type of the value for this NumberDataPoint.
+// Calling this function on zero-initialized NumberDataPoint will cause a panic.
+func (ms NumberDataPoint) Type() MetricValueType {
```

Comment:
On the list of "may fix": MetricValueType does not sound as the best name for the enum. Maybe NumberValueType?
PointValueType? Not sure, just open an issue, this name does not sound very clear to me, because I would ask why Histogram is not a value type.

Reply:
Makes sense. Filed an issue https://github.com/open-telemetry/opentelemetry-collector/issues/4819